src/qiskit_iqm/iqm_provider.py

```diff
@@ -26,16 +26,30 @@ class IQMProvider:
     Args:
         url: URL of the IQM Cortex server
         settings_path: path to the JSON settings file for the IQM backend
-        username: Username, if required by the IQM Cortex server. Can also be set in the ``IQM_SERVER_USERNAME``
-            environment variable.
-        api_key: API key, if required by the IQM Cortex server. Can also be set in the ``IQM_SERVER_API_KEY``
-            environment variable.
+
+    Keyword Args:
+        auth_server_url: URL of the user authentication server, if required by the IQM Cortex server.
+            Can also be set in the ``IQM_AUTH_SERVER`` environment variable.
+        username: Username, if required by the IQM Cortex server.
+            Can also be set in the ``IQM_AUTH_USERNAME`` environment variable.
+        password: Password, if required by the IQM Cortex server.
+            Can also be set in the ``IQM_AUTH_PASSWORD`` environment variable.
     """
-    def __init__(self, url: str, settings_path: str, username: str = None, api_key: str = None):
+    def __init__(
+            self,
+            url: str,
+            settings_path: str,
+            **user_auth_args  # contains keyword args auth_server_url, username, password
+    ):
         with open(settings_path, 'r', encoding='utf-8') as f:
-            self._client = IQMClient(url, json.loads(f.read()), username, api_key)
+            self._client = IQMClient(url, json.loads(f.read()), **user_auth_args)
 
     def get_backend(self) -> IQMBackend:
         """An IQMBackend instance associated with this provider.
         """
         return IQMBackend(self._client)
+
+    def close(self):
```

Comment:
Perhaps it is more convenient to have this method in `IQMBackend`, since `IQMProvider` is just used to generate backend objects and pass the client to them. We could do so that IQMProvider does not store client in the instance variable `self._client` at all. We could just save the arguments to ``__init__` as instance variables, and create the client object inside the `get_backend` method. Then `IQMProvider` will not need the close function at all, and the function could be moved to backend.

Reply:
Also, perhaps it could be renamed to `close_client` or something similar, since close is too general